File: src/main/java/levels/cafe/components/ground/Meal.java
package levels.cafe.components.ground;

import engine.graphics.Texture;
import engine.maths.Vector3f;

public class Meal extends AbstractGroundEntity {

    public Meal() {
        texture = generateTexture();
    }

    @Override
    protected Texture generateTexture() {
        return new Texture("res/room/meals/objekt_food_0001.png");
    }
}

File: src/main/java/levels/cafe/components/ground/Meal.kt
package levels.cafe.components.ground

import engine.graphics.Texture
import engine.graphics.TextureLoader
import engine.maths.Vector3f

class Meal : AbstractGroundEntity() {
    init {
        texture = generateTexture()
    }

    override fun generateTexture(): Texture {
        return TextureLoader.getTexture("res/room/meals/objekt_food_0001.png")
    }
}
